demos/BMW_demo/bmw_open.js

```diff
@@ -0,0 +1,170 @@
+ * @param {Array} parties - the parties to broadcast the share to.
+ * @param {number|string} op_id - a unique operation id, used to tag outgoing messages.
+ *
+ */
+
+/*
+
+function ooo(jiff,ls) {
+  var re=ls[1];
+  for (var i=2;i<=Object.keys(ls).length;i++) {
+    re=re^ls[i];
+
+  }
+  return re;
+
+}
+
+function recon_ls(jiff,ls) {
+  var re=ls[0];
+  for (var i=1;i<Object.keys(ls).length;i++) {
+    re=re^ls[i];
+
+  }
+  return re;
+
+}
+*/
+function recon_ls(jiff,ls) {
+  var re=ls[0];
+  for (var i=1;i<Object.keys(ls).length;i++) {
+    re=re^ls[i];
+
+  }
+  return re;
+
+
+}
+
+function bmw_reconstruct(jiff,shares) {
+  //console.log("inhelp");
+  //console.log(shares);
+  var ls=[];
+  for (let [key] of Object.keys(shares)) {
+    ls.push(shares[key]['value']);
+
+  }
+  //console.log(ls);
+  return recon_ls(jiff,ls);
+
+
+
+}
+
+var jiff_broadcast = function (jiff, share, parties, op_id) {
+  for (var index = 0; index < parties.length; index++) {
+    var i = parties[index]; // Party id
+    if (i === jiff.id) {
+      jiff.handlers.receive_open({ party_id: i, share: share.value, op_id: op_id, Zp: share.Zp });
+      continue;
+    }
+
+    // encrypt, sign and send
+    var msg = {party_id: i, share: share.value, op_id: op_id, Zp: share.Zp};
+    msg = jiff.hooks.execute_array_hooks('beforeOperation', [jiff, 'open', msg], 2);
+    msg['share'] = jiff.hooks.encryptSign(jiff, msg['share'].toString(), jiff.keymap[msg['party_id']], jiff.secret_key);
+    jiff.socket.safe_emit('open', JSON.stringify(msg));
+  }
+};
+
+module.exports = {
+  /**
+   * Open up the given share to the participating parties.
+   * @function jiff_open
+   * @ignore
+   * @param {module:jiff-client~JIFFClient} jiff - the jiff client instance.
+   * @param {module:jiff-client~JIFFClient#SecretShare} share - the share of the secret to open that belongs to this party
+   * @param {Array<number|string>} [parties=all_parties] - an array with party ids of receiving parties
+   * @param {string|number} [op_id=auto_gen()] - the operation id to be used to tag outgoing messages
+   * @returns {?promise} a (JQuery) promise to the open value of the secret, or null if the calling party is not a receiving party
+   *
+   */
+  bmw_jiff_open: function (jiff, share, parties, op_id) {
+
+    var i;
+    if (!(share.jiff === jiff)) {
+      throw 'share does not belong to given instance';
+    }
+
+    // Default values
+    if (parties == null || parties === []) {
+      parties = [];
+      for (i = 1; i <= jiff.party_count; i++) {
+        parties.push(i);
+      }
+    } else {
+      jiff.helpers.sort_ids(parties);
+    }
+
+    // If not a receiver nor holder, do nothing
+
+    if (share.holders.indexOf(jiff.id) === -1 && parties.indexOf(jiff.id) === -1) {
+      return null;
+    }
+
+
+    // Compute operation ids (one for each party that will receive a result
+    if (op_id == null) {
+      op_id = jiff.counters.gen_op_id2('open', parties, share.holders);
```

Comment:
I think we should change the op_id passed here ('open') to 'GMW_open' to avoid any confusion in the other functions. Ditto for any other GMW related functions that create op_ids.